php-frontend/src/main/java/org/sonar/php/tree/impl/lexical/InternalSyntaxToken.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.tree.impl.lexical;
+
+import com.sonar.sslr.api.TokenType;
+import org.sonar.php.parser.PHPTokenType;
+import org.sonar.php.tree.impl.PHPTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxTrivia;
+import org.sonar.plugins.php.api.visitors.TreeVisitor;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class InternalSyntaxToken extends PHPTree implements SyntaxToken {
+
+  private final Kind KIND;
+
+  private List<SyntaxTrivia> trivias;
+  private int startIndex;
+  private final int line;
+  private final int column;
+  private final String value;
+  private final boolean isEOF;
+
+  public InternalSyntaxToken(int line, int column, String value, List<SyntaxTrivia> trivias, int startIndex, boolean isEOF, TokenType type) {
+    this.value = value;
+    this.line = line;
+    this.column = column;
+    this.trivias = trivias;
+    this.startIndex = startIndex;
+    this.isEOF = isEOF;
+
+    if (type == PHPTokenType.INLINE_HTML) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)